external/vcm/vcm/xarray_loaders.py

```diff
@@ -112,3 +113,62 @@ def dump_nc(ds: xr.Dataset, f: BinaryIO):
         ds.to_netcdf(url, engine="h5netcdf")
         with open(url, "rb") as tmp1:
             shutil.copyfileobj(tmp1, f)
+
+
+def to_json(ds, filename, mode="w"):
+    """Write an xarray Dataset to JSON.
+
+    An important note about this function is that it does not support encoding
+    all datatypes supported by xarray.  It will raise an error if a variable
+    in the dataset is of a type that has not been tested.
+
+    Args:
+        ds: xr.Dataset
+        filename: str
+        mode: str (default w)
+    """
+    for name, var in ds.variables.items():
+        kind = var.dtype.kind
+        if kind not in "biufSU":
+            raise NotImplementedError(
+                f"It is not currently possible to serialize data of kind "
+                f"{kind!r}, the kind of variable {name}, to JSON in vcm."
+            )
+    with open(filename, mode) as file:
+        json.dump(ds.compute().to_dict(), file)
```

Comment:
Not sure if we want to add `indent=4` here or not (do we prefer single line JSON files or O(1000)-line JSON files?).

Reply:
Maybe could be an option, or you could just pass generic `kwargs` to `json.dump`? Probably for the reference test data better to have it be compact.

Reply:
Yeah, I agree no sense adding a huge amount of lines to the diff here.  I went ahead and implemented your suggestion.